src/Static.jl

```diff
@@ -430,6 +430,9 @@ Base.any(::Tuple{Vararg{True}}) = true
 Base.any(::Tuple{Vararg{Union{True, False}}}) = true
 Base.any(::Tuple{Vararg{False}}) = false
 
+Base.real(@nospecialize(x::StaticInteger)) = x
```

Comment:
```suggestion
Base.real(@nospecialize(x::StaticNumber)) = x
Base.real(@nospecialize(T::Type{<:StaticNumber})) = eltype(T)
```

This should avoid issues with type promotion.